src/rust/ide/view/src/project.rs

```diff
@@ -196,10 +178,18 @@ impl View {
                         }
                     } else {
                         model.searcher.hide();
-                        model.searcher.set_entries(list_view::entry::AnyModelProvider::default());
+                        model.searcher.clear_suggestions();
                     }
-                }
-            ));
+                })
+            );
+            _eval <- graph.outputs.node_position_set.map2(&graph.outputs.edited_node,
+                f!([searcher_left_top]((node_id,position),edited_node_id) {
+                    if edited_node_id.contains(node_id) {
+                        let new_left_top = Model::searcher_left_top_under_node_at(*position);
+                        searcher_left_top.set_target_value(new_left_top);
+                    }
```

Comment:
by reading the code I dont understand what it does. Could you refactor it to well named function instead please?